xarray/tests/test_plot.py

```diff
@@ -1812,6 +1820,109 @@ def test_wrong_num_of_dimensions(self):
             self.darray.plot.line(row='row', hue='hue')
 
 
+@requires_matplotlib
+class TestDatasetScatterPlots(PlotTestCase):
+    @pytest.fixture(autouse=True)
+    def setUp(self):
+        das = [DataArray(np.random.randn(3, 3, 4, 4),
+                         dims=['x', 'row', 'col', 'hue'],
+                         coords=[range(k) for k in [3, 3, 4, 4]])
+               for _ in [1, 2]]
+        ds = Dataset({'A': das[0], 'B': das[1]})
+        ds.hue.name = 'huename'
+        ds.hue.attrs['units'] = 'hunits'
+        ds.x.attrs['units'] = 'xunits'
+        ds.col.attrs['units'] = 'colunits'
+        ds.row.attrs['units'] = 'rowunits'
+        ds.A.attrs['units'] = 'Aunits'
+        ds.B.attrs['units'] = 'Bunits'
+        self.ds = ds
+
+    def test_facetgrid_shape(self):
+        g = self.ds.plot.scatter(x='A', y='B', row='row', col='col')
+        assert g.axes.shape == (len(self.ds.row), len(self.ds.col))
+
+        g = self.ds.plot.scatter(x='A', y='B', row='col', col='row')
+        assert g.axes.shape == (len(self.ds.col), len(self.ds.row))
+
+    def test_default_labels(self):
+        g = self.ds.plot.scatter('A', 'B', row='row', col='col', hue='hue')
+
+        # Top row should be labeled
+        for label, ax in zip(self.ds.coords['col'].values, g.axes[0, :]):
+            assert substring_in_axes(str(label), ax)
+
+        # Bottom row should have name of x array name and units
+        for ax in g.axes[-1, :]:
+            assert ax.get_xlabel() == 'A [Aunits]'
+
+        # Leftmost column should have name of y array name and units
+        for ax in g.axes[:, 0]:
+            assert ax.get_ylabel() == 'B [Bunits]'
+
+    def test_axes_in_faceted_plot(self):
+        with pytest.raises(ValueError):
+            self.ds.plot.scatter(x='A', y='B', row='row', ax=plt.axes())
+
+    def test_figsize_and_size(self):
+        with pytest.raises(ValueError):
+            self.ds.plot.scatter(x='A', y='B', row='row', size=3, figsize=4)
+
+    @pytest.mark.parametrize('x, y, hue_style, add_guide', [
+        ('A', 'B', 'something', True),
+        ('A', 'B', 'discrete', True),
+        ('A', 'B', None, True),
+        ('A', 'The Spanish Inquisition', None, None),
+        ('The Spanish Inquisition', 'B', None, True)])
+    def test_bad_args(self, x, y, hue_style, add_guide):
+        with pytest.raises(ValueError):
+            self.ds.plot.scatter(x, y, hue_style=hue_style,
+                                 add_guide=add_guide)
+
+    @pytest.mark.xfail(reason=['datetime, timedelta64 hue variable'
```

Comment:
I think needs to be a string, not a list.

Reply:
Thanks! let's try that.